app/src/main/java/com/naman14/timberx/ui/dialogs/CreatePlaylistDialog.kt

```diff
@@ -16,63 +16,63 @@ package com.naman14.timberx.ui.dialogs
 
 import android.app.Dialog
 import android.os.Bundle
-import android.widget.Toast
-
-import com.afollestad.materialdialogs.MaterialDialog
-import com.naman14.timberx.models.Song
-
 import androidx.annotation.NonNull
 import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import com.afollestad.materialdialogs.MaterialDialog
 import com.afollestad.materialdialogs.input.input
+import com.naman14.timberx.models.Song
 import com.naman14.timberx.util.MusicUtils
+import com.naman14.timberx.util.extensions.toast
 
 class CreatePlaylistDialog : DialogFragment() {
-
-    var callback: () -> Unit? = {
-        null
-    }
-
-    @NonNull
-    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-        return MaterialDialog(activity!!).show {
-            title(text = "Create new playlist")
-            positiveButton(text = "Create")
-            negativeButton(text = "Cancel")
-            input(hint = "Enter playlist name", callback = { dialog, text ->
-                val songs = arguments?.getLongArray("songs")
-
-                val playistId = MusicUtils.createPlaylist(activity!!, text.toString())
-
-                if (playistId.toInt() != -1) {
-                    if (songs != null && songs.isNotEmpty())
-                        MusicUtils.addToPlaylist(activity!!, songs, playistId)
-                    else
-                        Toast.makeText(activity, "Created playlist", Toast.LENGTH_SHORT).show()
-                    callback()
-                } else {
-                    Toast.makeText(activity, "Unable to create playlist", Toast.LENGTH_SHORT).show()
-                }
-            })
-        }
+    interface PlaylistCreatedCallback {
+        fun onPlaylistCreated()
     }
 
     companion object {
-        @JvmOverloads
-        fun newInstance(song: Song? = null): CreatePlaylistDialog {
+        private const val TAG = "CreatePlaylistDialog"
+
+        fun <T> show(parent: T, song: Song? = null) where T : Fragment, T : PlaylistCreatedCallback {
             val songs: LongArray
             if (song == null) {
                 songs = LongArray(0)
             } else {
                 songs = LongArray(1)
                 songs[0] = song.id
             }
-            return newInstance(songs)
+            show(parent, songs)
         }
 
-        fun newInstance(songList: LongArray): CreatePlaylistDialog {
-            return CreatePlaylistDialog().apply {
+        fun <T> show(parent: T, songList: LongArray) where T : Fragment, T : PlaylistCreatedCallback {
+            val dialog = CreatePlaylistDialog().apply {
                 arguments = Bundle().apply { putLongArray("songs", songList) }
             }
+            dialog.show(parent.childFragmentManager, TAG)
```

Comment:
Using `childFragmentManager` makes it so that `parentFragment` below returns the Fragment passes as a parameter to `show()`.

Reply:
I would think you could use regular `activity.supportFragmentManager` in conjunction with `dialog.setTargetFragment(parent)` (and `getTargetFragment()`).